airflow/ui/src/components/TaskInstanceTooltip.tsx

```diff
@@ -0,0 +1,64 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Box, Text } from "@chakra-ui/react";
+
+import type { TaskInstanceResponse } from "openapi/requests/types.gen";
+import Time from "src/components/Time";
+import { Tooltip } from "src/components/ui";
+
```

Comment:
Let's define props like so:
```
type Props = {
  readonly taskInstance: TaskInstanceResponse
} & TooltipProps;

Then we can use `children` instead of `child` and just wrap this tooltip around any component like `<TaskInstanceTooltip taskInstance={ti}><MyJsxComponent /></TaskInstanceTooltip>`

Reply:
Thanks TIL about pattern using `props.children` It seems `content` is required in `Tooltip` which I construct but using `Props & TooltipProps` causes typescript to expect `content` to be passed. I declared `children` explicitly inside `Props` and passed it on.

```
src/pages/DagsList/Dag/Tasks/TaskCard.tsx:72:12 - error TS2741: Property 'content' is missing in type '{ children: Element; taskInstance: TaskInstanceResponse; }' but required in type '{ content: ReactNode; contentProps?: TooltipContentProps | undefined; disabled?: boolean | undefined; portalled?: boolean | undefined; portalRef?: RefObject<...> | undefined; showArrow?: boolean | undefined; }'.

72           <TaskInstanceTooltip taskInstance={taskInstances[0]}>
              ~~~~~~~~~~~~~~~~~~~

  src/components/ui/Tooltip.tsx:23:3
    23   content: React.ReactNode;
         ~~~~~~~
    'content' is declared here.
```

Reply:
Oh we can omit content `& Omit<TooltipProps, "content">`

Reply:
Thanks, done.